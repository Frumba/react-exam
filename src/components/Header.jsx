import React, { Component } from 'react';
import PropTypes from 'prop-types';

import JediModal from './JediModal';

class Header extends Component {
  constructor() {
    super();

    this.state = {
      modalIsOpen: false
    };

    this.openModal = this.openModal.bind(this);
    this.closeModal = this.closeModal.bind(this);
  }

  openModal() {
    this.setState({ modalIsOpen: true });
  }

  closeModal() {
    this.setState({ modalIsOpen: false });
  }

  render() {
    const { modalIsOpen } = this.state;
    const { dispatch } = this.props;

    return (
      <div className="header">
        <h1 className="title">Jedis handler</h1>
        <button className="jedi-open-modal" onClick={this.openModal} type="button">
          Add Jedi
        </button>
        <JediModal dispatch={dispatch} isOpen={modalIsOpen} onRequestClose={this.closeModal} className="modal" />
      </div>
    );
  }
}

Header.propTypes = {
  dispatch: PropTypes.func.isRequired
};

export default Header;
