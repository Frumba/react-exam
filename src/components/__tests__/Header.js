import React from 'react';
import renderer from 'react-test-renderer';

import Header from '../Header';

jest.mock('../JediModal', () => 'jediModal');

it('Header render normally', () => {
  const tree = renderer.create(<Header dispatch={() => {}} />).toJSON();

  expect(tree).toMatchSnapshot();
});
