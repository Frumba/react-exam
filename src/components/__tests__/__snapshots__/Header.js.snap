// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render normally 1`] = `
<div
  className="header"
>
  <h1
    className="title"
  >
    Jedis handler
  </h1>
  <button
    className="jedi-open-modal"
    onClick={[Function]}
    type="button"
  >
    Add Jedi
  </button>
  <jediModal
    className="modal"
    dispatch={[Function]}
    isOpen={false}
    onRequestClose={[Function]}
  />
</div>
`;
